resources/js/Components/Layout/AuthOverlay/AuthOverlay.blocks.tsx

```diff
@@ -107,12 +108,21 @@ export const ConnectingWallet = ({ signing }: { signing: boolean }): JSX.Element
     );
 };
 
+const handleBackClick = (): void => {
+    if (!isTruthy(document.referrer)) {
+        window.location.href = "/";
+    } else {
+        window.history.back();
```

Comment:
I am not sure if we should blindly redirect user back, yes, that is a very minor possibility but a website might have a link to `/settings` or any route that requires signing - when we call `window.history.back();` it will redirect user to the referrer while we might want to keep user on the `dashbrd` by just redirecting it to the home page.

Reply:
it will be useful if we can look into handling this more elegantly yes. If it's a page on dashbrd then back should go back to the previous page, but otherwise home as @shahin-hq suggests.